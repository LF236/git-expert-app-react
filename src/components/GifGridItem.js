import React from 'react'
import PropTypes from 'prop-types';
const GifGridItem = ({ url, title }) => {
    return (
        <div className="gif-grid-item animate__fadeIn">
            <img src={ url } title={ title }/>
            <p>{ title }</p>
        </div>
    )
}

GifGridItem.propTypes = {
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
}

export default GifGridItem;