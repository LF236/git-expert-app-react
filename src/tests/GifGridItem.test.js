import React from 'react';
import { shallow } from "enzyme";
import GifGridItem from "../components/GifGridItem";

describe( 'Prueba con el componente <GifGridItem />', () => {
    const title = 'Un titulo';
    const url = 'http://localhost/img.jpg';
    const wrapper = shallow( <GifGridItem title={ title } url={ url }/> );
    test( 'Debe mostrar el componente correctamente renderizado', () => {
        expect( wrapper ).toMatchSnapshot();
    } );

    test( 'El componente renderizado debe tener un párrafo con el title', () => {
        const p = wrapper.find( 'p' );
        expect( p.text().trim() ).toBe( title );
    } );

    test( 'El componente renderizado debe tener la imágen igual al url y el alt de los props', () => {
        const img = wrapper.find( 'img' );
        expect( img.prop( 'src' ) ).toBe( url );
        expect( img.prop( 'title' ) ).toBe( title );
    } );

    test( 'El <div> principal debe tener la case animate__fadeIn', () => {
        const div = wrapper.find( 'div' );
        let clasessDiv = div.prop( 'className' );
        expect( clasessDiv.includes( 'animate__fadeIn' ) ).toBe( true );
    } );
} );