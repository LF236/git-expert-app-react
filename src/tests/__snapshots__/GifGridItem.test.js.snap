// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba con el componente <GifGridItem /> Debe mostrar el componente correctamente renderizado 1`] = `
<div
  className="gif-grid-item animate__fadeIn"
>
  <img
    src="http://localhost/img.jpg"
    title="Un titulo"
  />
  <p>
    Un titulo
  </p>
</div>
`;
